Fix issue #9

happycollision opened the pull request:
This will keep the `validation_errors` helper from being less-than-helpful.

happycollision committed:
Add specs for validation_errors

Tests the current, desired behavior
Also displays the current, unwanted behavior

happycollision committed:
Fix issue #9

It's the least I can do for opening it in the first place.

richmolj commented on lib/jsonapi_spec_helpers/helpers.rb:
```diff
@@ -56,6 +56,7 @@ def json_ids(integers = false)
 
     def validation_errors
       @validation_errors ||= {}.tap do |errors|
+        return if json['errors'].nil?
```
Could we change this like to

```ruby
errors = json['errors'] || []
errors.each do |e|
```

That way the method always returns a hash, and something like `validation_errors[:foo]` won't fail

happycollision replied:
The `errors` variable is already taken as the `self` of `{}.tap`. But yes, I have made that change by using a different variable.

I am more familiar with Javascript than Ruby, so I thought returning would still return—wait. I literally just thought of a better way as I was writing my explanation of what I did...

happycollision replied:
Okay. It is fixed below.

My heart knows Ruby better than Javascript, but my brain knows Javascript better than Ruby.

richmolj reviewed (commented):
Thanks for raising and addressing the issue @happycollision - had one small comment then would love to merge.

happycollision committed:
Add new spec for validation_errors

This extra expectation ensures that `validation_errors[:any_key]` also
won't throw an error.

happycollision committed:
Make the new spec pass

happycollision commented:
I did a fixup to the last commit when I realized I had a better solution where we didn't need a new local variable.

richmolj commented:
Thanks for the help @happycollision, much appreciated ❤️ ! LGTM 👍

richmolj commented:
Released in 0.4.7
